From 9eae71dcd60834f753264ddde13f37993f8b5556 Mon Sep 17 00:00:00 2001
From: Dan Callaghan <dcallagh@redhat.com>
Date: Mon, 7 Jan 2013 09:37:31 +1000
Subject: [PATCH 1/2] build shared library

---
 Makefile | 29 +++++++++++++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 3086e92..a1a906f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,19 @@
-all:  linenoise_example linenoise_utf8_example linenoise_cpp_example
+
+MAJOR_VERSION = 0
+EXTRA_VERSION = .0.0
+SONAME = liblinenoise.so.$(MAJOR_VERSION)
+LIB = $(SONAME)$(EXTRA_VERSION)
+
+export CFLAGS ?= -Os -g
+CFLAGS += -Wall -fpic -DUSE_UTF8
+LIBDIR ?= /usr/lib
+INCLUDEDIR ?= /usr/include
+
+.PHONY: all
+all: $(LIB) linenoise_example linenoise_utf8_example linenoise_cpp_example
+
+$(LIB): linenoise.o utf8.o
+	$(CC) $(CFLAGS) -shared -Wl,-soname,$(SONAME) $(LDFLAGS) -o $@ $^
 
 linenoise_example: linenoise.h linenoise.c example.c
 	$(CC) -Wall -W -Os -g -o $@ linenoise.c example.c
@@ -9,5 +24,15 @@ linenoise_utf8_example: linenoise.c utf8.c example.c
 linenoise_cpp_example: linenoise.h linenoise.c
 	g++ -Wall -W -Os -g -o $@ linenoise.c example.c
 
+.PHONY: clean
 clean:
-	rm -f linenoise_example linenoise_utf8_example linenoise_cpp_example *.o
+	rm -f $(LIB) linenoise_example linenoise_utf8_example linenoise_cpp_example *.o
+
+.PHONY: install
+install: $(LIB)
+	install -m 0755 -d $(DESTDIR)$(INCLUDEDIR)
+	install -m 0644 linenoise.h $(DESTDIR)$(INCLUDEDIR)
+	install -m 0755 -d $(DESTDIR)$(LIBDIR)
+	install -m 0755 $(LIB) $(DESTDIR)$(LIBDIR)
+	/sbin/ldconfig -n $(DESTDIR)$(LIBDIR)
+	ln -s $(LIB) $(DESTDIR)$(LIBDIR)/liblinenoise.so
-- 
1.7.11.7

